### DISCUSSION TITLE
Only enable DKIM signing if a message has a specific header

### DISCUSSION THREAD

**Discussion by nguyenkims** [rspamd/rspamd#3989] [Q&A] [answered]
Hi I'd like to know if there's a way to config Rspamd to only sign an outbound message if it contains a specific header? The reason is that we don't want to add a DKIM signature for some domains that don't have the DNS properly set up. 

We have tried using `path_map` that points to an URL that returns a list of domains to sign but 
- this approach is not efficient as we have thousands of domains.
- the highest check frequency seems to be every 5 minutes which is too slow for us as in the meantime, a lot of emails might have the DKIM signature wrongly added.

On https://rspamd.com/doc/configuration/settings.html, it seems `request_header` can be the solution here but I'm not sure.

Thanks!

**Comment by fatalbanana** [comment, marked as answer]
You could add a condition for `DKIM_SIGNED` symbol in Lua:

~~~lua
-- /etc/rspamd/rspamd.local.lua
rspamd_config:add_condition('DKIM_SIGNED', function(task) return task:has_header('Want-Signing') end)
~~~

**Reply by nguyenkims** [reply]
Thanks a lot! This is indeed what we are looking for.
